java/vector/src/main/java/org/apache/arrow/vector/VectorLoader.java

```diff
@@ -17,35 +17,90 @@
  */
 package org.apache.arrow.vector;
 
-import static com.google.common.base.Preconditions.checkArgument;
-
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-
+import com.google.common.collect.Iterators;
+import io.netty.buffer.ArrowBuf;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.complex.DictionaryVector;
+import org.apache.arrow.vector.schema.ArrowDictionaryBatch;
 import org.apache.arrow.vector.schema.ArrowFieldNode;
 import org.apache.arrow.vector.schema.ArrowRecordBatch;
 import org.apache.arrow.vector.schema.VectorLayout;
+import org.apache.arrow.vector.types.Dictionary;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.Types.MinorType;
+import org.apache.arrow.vector.types.pojo.ArrowType;
 import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
 
-import com.google.common.collect.Iterators;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
 
-import io.netty.buffer.ArrowBuf;
+import static com.google.common.base.Preconditions.checkArgument;
 
 /**
  * Loads buffers into vectors
  */
-public class VectorLoader {
+public class VectorLoader implements AutoCloseable {
+
   private final VectorSchemaRoot root;
+  private final Map<Long, FieldVector> dictionaryVectors = new HashMap<>();
 
   /**
-   * will create children in root based on schema
-   * @param schema the expected schema
-   * @param root the root to add vectors to based on schema
+   * Creates a vector loader
+   *
+   * @param schema schema
+   * @param allocator buffer allocator
    */
-  public VectorLoader(VectorSchemaRoot root) {
-    super();
-    this.root = root;
+  public VectorLoader(Schema schema, BufferAllocator allocator) {
```

Comment:
centralized into `VectorSchemaRoot`. What update needs to be made to the JSON format? I haven't added dictionaries there if that's what you're asking